File: LeetCode Solutions/June 2021 LeetCoding Challenge/Pascal's Triangle/Pascal's Triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> firstNumRows = new ArrayList<List<Integer>>(numRows);
        for (int i=0; i<numRows; i++){
            List<Integer> row = new ArrayList<>(i+1);
            while (row.size() <= i){
                row.add(1);
            }
            int mid = i >> 1;
            for (int j=1; j<=mid; j++){
                int val = firstNumRows.get(i-1).get(j-1) + firstNumRows.get(i-1).get(j);
                row.set(j, val);
                row.set(row.size()-j-1, val);
            }
            firstNumRows.add(row);
        }
        return firstNumRows;
    }
}

File: LeetCode Solutions/July 2022 LeetCoding Challenge/Pascal's Triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> firstNumRows(numRows);
        for (int i = 0; i < numRows; i++) {
            vector<int> row(i+1, 1);
            int mid = i >> 1;
            for (int j = 1; j <= mid; j++) {
                int val = firstNumRows[i-1][j-1] + firstNumRows[i-1][j];
                row[j] = val;
                row[row.size()-j-1] = val;
            }
            firstNumRows[i] = row;
        }
        return firstNumRows;
    }
};